yt_dlp/extractor/nosnl.py

```diff
@@ -37,59 +43,84 @@ class NOSNLArticleIE(InfoExtractor):
+            yield {
+                'id': str(data_process['id']),
+                'title': data_process.get('title'),
+                'description': data_process.get('description'),
+                'formats': formats,
+                'subtitles': subtitle,
+                'duration': parse_duration(data_process.get('duration')),
+                'thumbnails': [{
+                    'url': traverse_obj(image, ('url', ...), get_all=False),
+                    'width': image.get('width'),
+                    'height': image.get('height')
+                } for image in traverse_obj(data_process, ('imagesByRatio', ...))[0]],
+            }
+
+        elif data_process.get('type') == 'audio':
+            yield {
+                'id': str(data_process['id']),
+                'title': data_process.get('title'),
+                'url': traverse_obj(data_process, ('media', 'src')),
+                'ext': 'mp3',
+            }
 
-            elif item.get('type') == 'audio':
-                yield {
-                    'id': str(item['id']),
-                    'title': item.get('title'),
-                    'url': traverse_obj(item, ('media', 'src')),
-                    'ext': 'mp3',
-                }
+    def _entries(self, nextjs_json, display_id):
+        if isinstance(nextjs_json, dict):
+            yield from self._get_video_data(nextjs_json, display_id)
+        else:
+            for item in nextjs_json:
+                yield from self._get_video_data(item, display_id)
 
     def _real_extract(self, url):
-        display_id = self._match_valid_url(url).group('display_id')
+        site_type, display_id = self._match_valid_url(url).group('type', 'display_id')
         webpage = self._download_webpage(url, display_id)
 
         nextjs_json = self._search_nextjs_data(webpage, display_id)['props']['pageProps']['data']
         return {
             '_type': 'playlist',
-            'entries': self._entries(nextjs_json, display_id),
+            'entries': self._entries(
+                nextjs_json['video'] if site_type == 'video' else nextjs_json['items'], display_id),
             'id': str(nextjs_json['id']),
             'title': nextjs_json.get('title') or self._html_search_meta(['title', 'og:title', 'twitter:title'], webpage),
             'description': (nextjs_json.get('description')
                             or self._html_search_meta(['description', 'twitter:description', 'og:description'], webpage)),
             'tags': nextjs_json.get('keywords'),
             'modified_timestamp': parse_iso8601(nextjs_json.get('modifiedAt')),
             'thumbnail': nextjs_json.get('shareImageSrc') or self._html_search_meta(['og:image', 'twitter:image'], webpage),
-            'timestamp': parse_iso8601(nextjs_json.get('publishedAt'))
+            'timestamp': parse_iso8601(nextjs_json.get('publishedAt')),
+            'categories': try_call(lambda: traverse_obj(nextjs_json, ('categories', ..., 'label')))
```

Comment:
```suggestion
            'categories': traverse_obj(nextjs_json, ('categories', ..., 'label')),
```
traverse_obj never raises